src/common/exception/GraknClientException.jl

```diff
@@ -2,13 +2,35 @@
 
 mutable struct GraknClientException <: Exception
     error_message::T where {T<:Union{<:AbstractGeneralError,<:Nothing}}
-    params
+    params::Optional{Tuple}
     individual_message::Union{Nothing, String}
-    cause::R where {R<:Exception}
+    cause::R where {R<:Union{Nothing, Exception}}
+end
+
+Base.show(io::IO, grakn_excption::GraknClientException) = Base.print(io,grakn_excption)
+function Base.print(io::IO, grakn_excption::GraknClientException)
+    err_message = string(grakn_excption.error_message)
+    joined_params = ""
+    if grakn_excption.params !== nothing
+        if !isempty(grakn_excption.params)
+            replace_item = "_error_item"
+            err_message = replace(err_message, replace_item=>string(grakn_excption.params[1]))
+            joined_params = join(grakn_excption.params, "\n")
+        end
+    end
+    str = "GraknClientException:
```

Comment:
Better to not build a string, just to then print it out.
That causes unnecessary allocations.
Yes, this method may not be called frequently, but it's good to get in good habits when you code :-)

Reply:
Just:
 ```
println(io, "TypeDBClientException:")
println(io, "message: ", err_message)
println(io, "params:    ", joined_params)
println(io, "remark:     ", exc.individual_message)
print(io,    "cause:       ")
print(io,    "exc.cause")